### PR TITLE
Fix/docker release

### LINKED ISSUE #160: 🐛 Remove $HOME Override in Dockerfile
## Overview

Currently in multiple places we override the `$HOME` env variable in dockerfiles. What this is doing essentially, is forcing the node operators who use our Dockerfiles to pass in a custom home directory upon running:
```
$ seda-chaind start --home <home_dir>
```
If `--home` is not passed, it defaults to the `$HOME` that we overrode prematurely in the Dockerfile, and the operator would see:
```
panic: open /seda-chain/.seda-chain/config/genesis.json: no such file or directory
```
`$HOME` should always be the default set by the operating system that the operator chooses to run `seda-chaind` on, typically `/root`. This is the default behavior by the sdk and we should respect that.

### LINKED ISSUE #198: 🐛 Broken Dockerfile and Docker Release
## 🐛 Bug Report

A few things have gone wrong since we decided to focus on non-Docker deployment. So this issue is to describe what's broken so we can fix at a future date.

## Steps to Reproduce

Try running the latest built docker image.

#### Code snippet to reproduce

```bash
docker run -d --name seda_node \
      --env-file seda.env \
      --volume ~/.seda-chain:/root/.seda-chain \
      --volume $(pwd)/seda.env:/seda-chain/.env \
      ghcr.io/sedaprotocol/node:latest start
```

#### Stack trace & error message

```bash
Initializing Node ...
Setting Up seda configuration
Error: error reading GenesisDoc at /seda-chain/.seda-chain/config/genesis.json: invalid 64-bit integer encoding "1", expected string
```

## Expected Behavior

Should work.

Here's what has to be done to fix it:
- [ ] The Dockerfile and scripts rely on the old path and binary name(well, once we merge #193  it should be fixed).
- [ ] It's an older version, so it doesn't like the Genesis file(I think this is the issue?). I.e. we need to do a new release.
- [ ] Possibly more that's blocked behind the above.

This blocks #160, or it doesn't fully work, but we can only test that if the docker container works.

## Your Environment

Doesn't matter since it's docker.

### PR DESCRIPTION
**Pull request by gluax**
## Motivation

To fix and update our released docker images.

## Explanation of Changes

1. Updated the caching rules for the docker build in the release CI.
2. Added cosmovisor to the container.
3. Simplified the docker stuff.
4. Removed the old scripts we used for the docker image.
5. Added docker section to the readme again.

## Testing

[Tested on my fork](https://github.com/gluax/seda-chain/actions/workflows/release.yml)

## Related PRs and Issues

Closes #160.
Closes #198

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,60 @@ rm -rf ~/.sedad || true
...
+For instructions how to run the node yourself as a normal node or a validator in [docker](https://www.docker.com/). 
+
+To pull the `docker` image check [here](#pulling-the-container).
+
+This section will go over:
+- [docker commands](#docker-commands)
+- Setting up [env variables](#env-variables-configuration) for `docker`
+<!-- - How to create a [key](#key-creation) -->
+
+<!-- We recommend looking at the commands section first so if you need to run `seda-chaind` commands to create a key you can do so from within docker. -->
```
Comment (mariocao) [inline README.md:122]: This hidden text can be removed, right?

File: README.md [hunk 2]
```diff
@@ -108,6 +108,60 @@ rm -rf ~/.sedad || true
...
+To start a SEDA chain node with `docker`(we recommend knowing the tool if you go this route):
+
+```bash
+docker run -d --name seda_node \
+-p 26656:26656 \
+-p 26657:26657 \
+-p 9090:9090 \
+--env MONIKER=moniker_here
+--env NETWORK=testnet
+ghcr.io/gluax/seda-chain:latest sedad start
```
Comment (mariocao) [inline README.md:139]: Change `ghcr.io` path :P

Comment (mariocao) [inline README.md:139]:
```suggestion
ghcr.io/sedaprotocol/seda-chain:latest sedad start
```

Comment (gluax) [reply README.md:139]: LOL thanks

File: README.md [hunk 3]
```diff
@@ -108,6 +108,57 @@ rm -rf ~/.sedad || true
 ./sedad start
 ```
 
+### Running the Node Yourself Dockerized
+
+For instructions how to run the node yourself as a normal node or a validator in [docker](https://www.docker.com/). 
+
+To pull the `docker` image check [here](#pulling-the-container).
```
Comment (hacheigriega) [inline README.md:115]: No link here?

**Review by hacheigriega — CHANGES_REQUESTED** [review]

**Review by mariocao — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (gluax) [reply README.md:115]: Hmm there are links no? One to link to the section and one to the docker website.

**Pull request merged**